mesonbuild/mtest.py

```diff
@@ -289,8 +290,8 @@ def _run_cmd(self, cmd):
         stdout = None
         stderr = None
         if not self.options.verbose:
-            stdout = subprocess.PIPE
-            stderr = subprocess.PIPE if self.options and self.options.split else subprocess.STDOUT
+            stdout = tempfile.TemporaryFile("w+", encoding='utf-8')
+            stderr = tempfile.TemporaryFile("w+", encoding='utf-8')
```

Comment:
This breaks stdout splitting. If it is set, stdout and stderr should go intermixed in one stream.

Reply:
I think i have this now resolved - however I ran into another problem that when a test uses color output, then the mtest.py thus raised a error because of none utf-8 characters, I fixed this now with changing the encoding. However - I am not sure, is this correct ?

Reply:
The output file should be opened in binary mode. The output is assumed to be raw bytes.

Reply:
Okay - thx for the help. Should be right now :)